File: packages/mobile-app-did/android/app/src/main/java/com/PortkeyApp/native_modules/PortkeyPackage.java
package com.PortkeyApp.native_modules;

import androidx.annotation.NonNull;

import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;

import java.util.ArrayList;
import java.util.List;

public class PortkeyPackage implements ReactPackage {
    @NonNull
    @Override
    public List<NativeModule> createNativeModules(@NonNull ReactApplicationContext reactApplicationContext) {
        return new ArrayList<NativeModule>() {{
            add(new AppLifeCycleModule(reactApplicationContext));
        }};
    }

    @NonNull
    @Override
    public List<ViewManager> createViewManagers(@NonNull ReactApplicationContext reactApplicationContext) {
        return new ArrayList<>();
    }
}

File: packages/mobile-app-did/android/app/src/main/java/com/PortkeyApp/native_modules/PortkeyPackage.kt
package com.PortkeyApp.native_modules

import com.PortkeyApp.compose.qr_code.QrViewManager
import com.PortkeyApp.native_modules.lifecycle.AppLifeCycleModule
import com.facebook.react.ReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ViewManager

class PortkeyPackage : ReactPackage {
    override fun createNativeModules(reactApplicationContext: ReactApplicationContext): List<NativeModule> {
        return mutableListOf(AppLifeCycleModule(reactApplicationContext))
    }

    override fun createViewManagers(reactApplicationContext: ReactApplicationContext): List<ViewManager<*, *>> {
        return mutableListOf(QrViewManager())
    }
}
